core/src/main/java/org/jdbi/v3/core/argument/BeanPropertyArguments.java

```diff
@@ -64,7 +65,16 @@ public BeanPropertyArguments(String prefix, Object bean)
 
                     try
                     {
-                        return ctx.findArgumentFor(getter.getGenericReturnType(), getter.invoke(bean));
+                        Type propertyType = getter.getGenericReturnType();
+                        Object propertyValue = getter.invoke(bean);
+                        Argument argument = ctx.findArgumentFor(propertyType, propertyValue)
```

Comment:
I think an explicit check that the argument is present is better than extracting an optional value to present value and then back to `Optional`.
```java
Optional<Argument> argument = ctx.findArgumentFor(propertyType, propertyValue);
if (argument.isPresent()) {
    throw new UnableToCreateStatementException(
       String.format("No argument factory registered for type [%s] for bean property [%s] on [%s]",
       propertyType, propertyName, bean), ctx);
}
return argument;
```